hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieKeyBasedFileGroupRecordBuffer.java

```diff
@@ -86,6 +88,10 @@ public void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpecO
   @Override
   public void processNextDataRecord(T record, Map<String, Object> metadata, Serializable recordKey) throws IOException {
     Pair<Option<T>, Map<String, Object>> existingRecordMetadataPair = records.get(recordKey);
+    if (existingRecordMetadataPair != null && existingRecordMetadataPair.getRight().containsKey(DELETE_FOUND_WITHOUT_ORDERING_VALUE)) {
```

Comment:
Add a comment here saying that because we process the logs in reverse order the delete we found earlier must have a later processing time and didn't have ordering value so we assume that it's definitely overwriting

Reply:
sg. Done.